torch/fx/passes/infra/partitioner.py

```diff
@@ -164,14 +164,17 @@ def dfs_iter_find_cycle(all_user_nodes: Set[Node]):
 
         def merge_single_node(node: Node, id: Optional[int]):
             def _update_partition_map(node: Node, id: int):
-                # Iterate through all the downstream nodes of this node and update the partition map
-                # to indicate that there is a path from the partition id of this node to the target
-                # partition id.
-                downstream_nodes = self.dependency_viewer.downstreams_of(node)
-                for curr_node in downstream_nodes:
-                    target_id = assignment.get(curr_node, None)
+                # Iterate through all the users of this node and update the partition map to indicate
+                # that there is a path from the partition id of this node to the target partition id.
+                for user_node in node.users:
+                    target_id = assignment.get(user_node, None)
                     if target_id is not None:
                         partition_map[id].add(target_id)
+                        partition_map[id].update(partition_map[target_id])
+                    else:
+                        assert not self.__is_node_supported(
+                            user_node
+                        ), "users which have not been traversed must be unsupported node"
```

Comment:
nit: Change error to `Encountered user node which has not been traversed yet. This should only happen if this is an unsupported node.`